### ISSUE TITLE
Complete Tree Capture Flow

### ISSUE THREAD

**Issue by Elforama** [Greenstand/treetracker-android#647]
The tree capture flow:
- 1 Click "track"
- 2 select a user
- 3 select a wallet
- 4 Click camera capture button
- 5 Wait for loading icon as GPS signal is captured
- 6 Picture is taken
- 7 Click "accept" to save the tree or "retake" to retake the photo (which brings you back to the camera screen)
- 8 If accept is clicked, the tree should be saved. 
- 9 Repeat from step 4 until done capturing trees
- 10 Exit tree capture flow to dashboard

Current State:
Steps 1 -> 7 seem to be working. Once a tree is saved it goes back the camera screen, but it's blank, and it seems like the app is frozen.

Task: Make sure steps 1 -> 10 all work. 

To make sure trees are user data is being saved you can use the app inspector in android studio. It will show you the contents of the DB.

**Comment by EngrTaofeek** [comment]
It doesn't go beyond 4 and 5. I tried both with my gps and on same result kept crashing. The convergence function in Locations.kt function kept ruturning null. I dodn't really seem to understand fully well the location implemenation usage.

**Comment by EngrTaofeek** [comment]
Yeah apparently it occurs when Convergence request times out. If the location isn't determined in a certain amount of time the convergence location returns null. Tried with another device and emulator it works well.
